<a href="https://colab.research.google.com/github/SethJensen76/Discrete-Mathematics-Spring-2022/blob/main/Project_Part_4_Counting.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project Part 4: CPSMA 3133 Discrete Mathematics 

## Counting

### Seth Jensen 244476




So I am going to continue on my Jacks, Queens, and Kings that I used for part 3.



In [10]:
JQK = ['Jacks','Queens','Kings']

## Counting

Since there are 3 types of cards, and multiple types of outcomes, I will be randomizing it 12 times. So it will be $3**12$

In [1]:
3**12

531441

Since there is no specific order, and there is replacement, we will use the following formula. For example, if there are $n$ objects and $k$ slots but order does not matter, the number of ways to represent this are:

$$
\binom{n+k-1}{k-1} = \binom{n+k-1}{n}
$$

In [8]:
import math

math.factorial(15)/(math.factorial(12)*math.factorial(3))

455.0

I feel like this is a pretty big difference than from up above!
455 is way less than the 531,441 computed earlier.
This is also with replacement.

In [11]:
from itertools import combinations_with_replacement

print(list(combinations_with_replacement(JQK,10)))

[('Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks'), ('Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Queens'), ('Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Kings'), ('Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Queens', 'Queens'), ('Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Queens', 'Kings'), ('Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Kings', 'Kings'), ('Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Queens', 'Queens', 'Queens'), ('Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Queens', 'Queens', 'Kings'), ('Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Queens', 'Kings', 'Kings'), ('Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Kings', 'Kings', 'Kings'), ('Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Jacks', 'Queens', 'Que

It has the correct number of elements, really long list of them though...

In [14]:
len(list(combinations_with_replacement(JQK,52)))

1431

This is interesting to me!

## Without replacement

Since I can't do all 15 with replacement, and order matters here, I will only be doing 3, which is all I can do!

In [20]:
print(math.factorial(3))


6


I'll do the rest of them as well...

In [21]:
from itertools import permutations
print(list(permutations(JQK,3)))

[('Jacks', 'Queens', 'Kings'), ('Jacks', 'Kings', 'Queens'), ('Queens', 'Jacks', 'Kings'), ('Queens', 'Kings', 'Jacks'), ('Kings', 'Jacks', 'Queens'), ('Kings', 'Queens', 'Jacks')]


When order doesn't matter, the following will happen:

In [22]:
math.factorial(3)/(math.factorial(3)*math.factorial(1))

1.0

This is interesting that it would equal out to 1, so that means that only the following can happen:

In [23]:
from itertools import combinations
print(list(combinations(JQK,3)))


[('Jacks', 'Queens', 'Kings')]


Alright, I'm done! I found this very interesting and fun. Please let me know if I did anything wrong, right, or both!